From 28d0147bded959b2c4d3eb1aa957452d5dbb0cc9 Mon Sep 17 00:00:00 2001
From: Fedora Kernel Team <kernel-team@fedoraproject.org>
Date: Mon, 20 Jun 2016 14:52:10 +0200
Subject: [PATCH 5/6] i915/fbc: Disable on HSW by default for now

Upstream: posted on dri-devel (and r-b'd)

Author:     cpaul@redhat.com <cpaul@redhat.com>
AuthorDate: Thu Jun 9 11:58:15 2016 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Thu Jun 9 15:43:07 2016 -0400

    i915/fbc: Disable on HSW by default for now

    >From https://bugs.freedesktop.org/show_bug.cgi?id=96461 :

    This was kind of a difficult bug to track down. If you're using a
    Haswell system running GNOME and you have fbc completely enabled and
    working, playing videos can result in video artifacts. Steps to
    reproduce:

    - Run GNOME
    - Ensure FBC is enabled and active
    - Download a movie, I used the ogg version of Big Buck Bunny for this
    - Run `gst-launch-1.0 filesrc location='some_movie.ogg' ! decodebin !
      glimagesink` in a terminal
    - Watch for about over a minute, you'll see small horizontal lines go
      down the screen.

    For the time being, disable FBC for Haswell by default.

    Signed-off-by: Lyude <cpaul@redhat.com>
    Reviewed-by: Paulo Zanoni <paulo.r.zanoni@intel.com>
    Cc: stable@vger.kernel.org
---
 drivers/gpu/drm/i915/intel_fbc.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_fbc.c b/drivers/gpu/drm/i915/intel_fbc.c
index 0f0492f..28f4407 100644
--- a/drivers/gpu/drm/i915/intel_fbc.c
+++ b/drivers/gpu/drm/i915/intel_fbc.c
@@ -823,8 +823,7 @@ static bool intel_fbc_can_choose(struct intel_crtc *crtc)
 {
 	struct drm_i915_private *dev_priv = crtc->base.dev->dev_private;
 	struct intel_fbc *fbc = &dev_priv->fbc;
-	bool enable_by_default = IS_HASWELL(dev_priv) ||
-				 IS_BROADWELL(dev_priv);
+	bool enable_by_default = IS_BROADWELL(dev_priv);
 
 	if (intel_vgpu_active(dev_priv->dev)) {
 		fbc->no_fbc_reason = "VGPU is active";
-- 
2.7.4

